daemon/container.go

```diff
@@ -261,45 +268,46 @@ func validateContainerConfig(config *containertypes.Config) error {
 	return validateHealthCheck(config.Healthcheck)
 }
 
-func validateHostConfig(hostConfig *containertypes.HostConfig) error {
+func validateHostConfig(hostConfig *containertypes.HostConfig) ([]string, error) {
 	if hostConfig == nil {
-		return nil
+		return nil, nil
 	}
 
+	var warnings []string
 	if hostConfig.AutoRemove && !hostConfig.RestartPolicy.IsNone() {
-		return errors.Errorf("can't create 'AutoRemove' container with restart policy")
+		return nil, errors.Errorf("can't create 'AutoRemove' container with restart policy")
```

Comment:
I'm slightly on the fence what's more robust here; usually, I'm a big fan of being explicit (return `nil` if there's nothing to return). However, in this case, the `warnings` array is something that's aggregated / added to in multiple places. Returning `nil` here makes it error-prone for situations where (e.g.) another check is added further up that appends to `warnings` (which we would discard if we return `nil` here).

So _probably_ returning `warnings` here is the right thing to do (i..e., always return whatever we collected until now, which may be "nothing")

```go
return warnings, errors.Errorf("can't create 'AutoRemove' container with restart policy")
```